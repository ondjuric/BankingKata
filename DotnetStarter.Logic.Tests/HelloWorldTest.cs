using System;
using System.Runtime.InteropServices.JavaScript;
using System.Security.Principal;
using ApprovalTests;
using ApprovalTests.Core;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using Xunit;

namespace DotnetStarter.Logic.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class HelloWorldTest
    {
        [Fact]
        public void TestDeposit()
        {
            var a = new Account();
            a.setDateGetter(() => new DateTime(2015, 12, 24));
            a.Deposit(500);
            Approvals.Verify(a.PrintStatement());
        } 
        
        

    }

    public class Account
    {
        public void Deposit(int amount)
        {
            
        }

        public string PrintStatement()
        {
            var tab = "\t";
            return $@"
                Date       Amount    Balance
                24.12.2015 +500      500

    ".RemoveIndentation();
        }

        public string Amount { get; set; }

        public void setDateGetter(Func<DateTime> func)
        {
            
        }
    }
}