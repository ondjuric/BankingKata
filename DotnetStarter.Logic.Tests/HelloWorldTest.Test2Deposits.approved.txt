Date       Amount    Balance
01.12.2015 +600      600
02.12.2015 +600      1200
